aptos-move/framework/aptos-names/sources/test_helper.move

```diff
@@ -69,7 +70,11 @@ module aptos_names::test_helper {
 
         let years = (time_helper::seconds_to_years(registration_duration_secs) as u8);
         if (option::is_none(&subdomain_name)) {
-            domains::register_domain(user, domain_name, years);
+            if (unrestricted_mint_enabled()) {
+                domains::register_domain(user, domain_name, years, vector::empty<u8>());
```

Comment:
If `register_domain` is supposed to not verify a signature, why is it even taking a signature parameter?

Reply:
I think we might want to re-design this more cleanly.

Reply:
I think we want to have a generic function that might or might not require a signature (if not, we can bypass it by passing an empty signature). And then in `register_domain_with_signature` we're enforcing it.

Reply:
I would recommend either one of the following, but not both:
1. Two separate API, one that requires & verifies a signature argument & one that doesn't, as I described [here](https://github.com/aptos-labs/aptos-core/pull/4889#discussion_r991561064)
2. One API that takes an optional signature and enforces the check based on `unrestricted_mint_enabled()`